src/dev/headless/request-handler-controller.ts

```diff
@@ -52,8 +54,13 @@ export class RequestHandlerController {
           await this.bundlePathPromise,
           "utf-8"
         )
+        const { middlewares } = this
         this.cachedWinterCGRuntime = new EdgeRuntime({
           initialCode: contents,
+          extend(context) {
+            context._injectedEdgeSpecMiddlewares = middlewares
+            return context
+          },
```

Comment:
Why can't we wrap the middleware at top level?

Reply:
sorry, what do you mean by top level?

my plan is to replace this PR by bundling a module service entrypoint by default, then allowing that entrypoint to accept middleware (probably will still have to inject using this method for the edge runtime)